app/workers/open_aq_import_measurements_worker.rb

```diff
@@ -18,13 +18,94 @@ class OpenAqImportMeasurementsWorker
-  # indexes in the database) but this seemed to be a simple and good enough solution.
-  sidekiq_options lock: :until_executed, on_conflict: :log
+  # Unfortunately, we can't use the `:until_executed` lock from sidekiq_unique_jobs because when the
+  # Sidekiq server is unexpectedly shut down while this worker is working, the lock will stay in
+  # Redis until someone manually removes it.
+  #
+  # Typically this problem is solved by making sure the lock stays either until the worker finishes
+  # or until a certain amount of time passes. [1]
+  #
+  # Unfortunately again, sidekiq_unique_jobs, unlike the enterprise version of Sidekiq, doesn't
+  # support such locking strategy. Instead, we have to hack it ourselves by using the
+  # `:until_expired` lock and Sidekiq's API to remove the lock when the worker finishes.
+  #
+  # `:until_expired` ensures that the lock expires after the time specified through
+  # `lock_expiration`. In case Sidekiq is shut down unexpectedly, the lock will eventually expire
+  # and Sidekiq will be able to enqueue this worker again.
+  #
+  # As [1] explains, this method doesn't guarantee uniqueness, but the alternative is having a
+  # system that requires a manual intervention after an unexpected restart/shutdown. [2] goes in
+  # detail as to why the timeout is required.
+  #
+  # Another unfortunate side effect of using `:until_expired` is that locks created by it aren't
+  # visible through the "Unique Digests" panel in Sidekiq dashboard as well as through
+  # `SidekiqUniqueJobs::Digests.all`. But the logs should still show info about skipping the worker
+  # because the unique digest being present. Attempting to enqueue the worker from console will
+  # return the error as well.
+  #
+  # One way to guarantee uniqueness for the concurrent inserts would be to use a different
+  # architecture (e.g. unique indexes in the database).
+  #
+  # [1] https://github.com/mperham/sidekiq/wiki/Ent-Unique-Jobs#use
+  # [2] https://redis.io/topics/distlock#why-failover-based-implementations-are-not-enough
+  sidekiq_options lock: :until_expired,
+                  on_conflict: :log,
+                  lock_expiration: 30.minutes
 
   def perform
+    started_at = Time.now
+
     return unless A9n.sidekiq_open_aq_import_measurements_enabled
 
     OpenAq::ImportMeasurements.new.call
+  ensure
+    remove_lock(started_at)
+  end
+
+  def remove_lock(started_at)
+    if !jid
+      # Skip lock removal logic in case the worker was executed synchronously (and thus has no jid).
+      logger.info "#{
+                    self.class.name
+                  } was executed synchronously. Skipping lock removal."
+      return
+    end
+
+    if Time.now - started_at < 5.seconds
+      # Sidekiq::Workers data is updated every 5 seconds.
+      # https://github.com/mperham/sidekiq/wiki/API#workers
+      logger.info 'Sleeping for 5 seconds to wait for Sidekiq workers data to refresh.'
+      sleep 5
+    end
```

Comment:
When I found out that this is the case I was like, okay, this is getting quite complex already. 🤕

But really there's no other way to approach it if we want to remove this lock manually from within the `perform` method. Within `perform`, we don't have any meta info about the payload of the worker other than `jid`. If we had `unique_digest` then I wouldn't need to do all this waiting.